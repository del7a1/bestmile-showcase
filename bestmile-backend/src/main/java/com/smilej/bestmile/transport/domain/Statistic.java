package com.smilej.bestmile.transport.domain;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Statistic {

    private long missionsCounter;
    private double totalPassengerCount;
    private double totalDistance;

}
