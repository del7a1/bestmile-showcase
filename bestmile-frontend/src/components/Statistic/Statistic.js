import React, { Component } from 'react';
import Popup from 'react-popup'
import axios from 'axios';

export default class Statistic extends Component {

  constructor() {
    super();
    this.showStatistics = this.showStatistics.bind(this);
  }

  showStatistics() {
    axios.get('')
      .then(response => {
        Popup.alert(response);
      })
  }

  render() {
    return (
      <div>
        <button onClick={this.showStatistics} value='Show Statistics'/>
        <Popup />
      </div>
    );
  }

}
