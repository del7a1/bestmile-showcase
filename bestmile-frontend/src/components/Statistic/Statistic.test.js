import React from 'react';
import { shallow } from 'enzyme';
import Statistic from './Statistic.js';

describe('Statistic', () => {

 it('Should be defined', () => {
   expect(Statistic).toBeDefined();
 });

 it('Should render correctly', () => {
   const wrapper = shallow(
     <Statistic />
   );
   expect(wrapper).toMatchSnapshot();
 });

 it('Should has defined value', () => {
   const wrapper = shallow(
     <Statistic />
   );
   expect(typeof(wrapper.find('button').getElement().props.value)).toBe('string');
   expect(wrapper.find('button').getElement().props.value).toEqual('Show Statistics');
 });

});
