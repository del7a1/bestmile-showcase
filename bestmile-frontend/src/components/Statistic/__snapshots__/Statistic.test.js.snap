// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistic Should render correctly 1`] = `
<div>
  <button
    onClick={[Function]}
    value="Show Statistics"
  />
  <Popup
    btnClass="mm-popup__btn"
    className="mm-popup"
    closeBtn={true}
    closeHtml={null}
    closeOnOutsideClick={true}
    defaultCancel="Cancel"
    defaultCancelKey="esc"
    defaultOk="Ok"
    defaultOkKey="enter"
    escToClose={true}
  />
</div>
`;
